### PR TITLE
Add org.kde.kst

### PR DESCRIPTION
**Pull request by ripxorip**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: org.kde.kst.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  - /mkspecs
+  - /share/aclocal
+  - /share/info
+  - /share/man
+modules:
+
+  - name: gsl
+    sources:
+      - type: archive
+        url: 'http://ftp.gnu.org/gnu/gsl/gsl-2.7.1.tar.gz'
```
Comment (hfiguiere) [inline org.kde.kst.yml:28]: https ?

File: org.kde.kst.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+runtime: org.kde.Platform
+runtime-version: 5.15-21.08
+sdk: org.kde.Sdk
+command: kst2
+finish-args:
+  - '--share=network'
+  - '--share=ipc'
+  - '--socket=x11'
+  - '--socket=wayland'
+  - '--filesystem=host'
```
Comment (hfiguiere) [inline org.kde.kst.yml:11]: that is a very broad filesystem permission

File: org.kde.kst.yml [hunk 3]
```diff
@@ -0,0 +1,99 @@
+app-id: org.kde.kst
+runtime: org.kde.Platform
+runtime-version: 5.15-21.08
+sdk: org.kde.Sdk
+command: kst2
+finish-args:
+  - '--share=network'
+  - '--share=ipc'
+  - '--socket=x11'
```
Comment (hfiguiere) [inline org.kde.kst.yml:9]: does it support wayland? if yes:
```suggestion
  - '--socket=fallback-x11'
```

otherwise remove the wayland socket.

File: org.kde.kst.yml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        x-checker-data:
+          type: anitya
+          project-id: 10354
+          stable-only: true
+          url-template: 'https://github.com/Unidata/netcdf-c/archive/v$version.tar.gz'
+
+  - name: netcdfcxx
+    sources:
+      - type: archive
+        url: 'ftp://ftp.unidata.ucar.edu/pub/netcdf/netcdf-cxx-4.2.tar.gz'
```
Comment (hfiguiere) [inline org.kde.kst.yml:80]: no https?

**Comment by hfiguiere** [comment]
bot, build org.kde.kst

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: org.kde.kst.yml (hunk 2 above)
Comment (ripxorip) [reply org.kde.kst.yml]: Fixed, most user will want to plot stuff exclusively from their home anyway

File: org.kde.kst.yml [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        sha256: 95ed6ab49a0ee001255eac4e44aacb5ca4ea96ba850c08337a3e4c9a0872ccd1
+
+  - name: kst
+    buildsystem: cmake
```
[the remark this answers is not in the document]
Comment (ripxorip) [reply org.kde.kst.yml]: Just tested and it works as extected, fixed

File: org.kde.kst.yml (hunk 4 above)
Comment (ripxorip) [reply org.kde.kst.yml:79]: This one was trickier could not find one with https

File: org.kde.kst.yml (hunk 3 above)
Comment (ripxorip) [reply org.kde.kst.yml]: Fixed

File: org.kde.kst.yml (hunk 1 above)
Comment (ripxorip) [reply org.kde.kst.yml]: Fixed

File: flathub.json [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
+	"only-arches": ["x86_64"],
```
Comment (hfiguiere) [inline flathub.json:2]: This is not the fix.

Either you take a more recent hdf5 that build out of the box, or you apply the usual "outdated configure" fix: adding a `shell` source that will do `cp -p /usr/share/automake-*/config.{sub,guess} .`.

File: org.kde.kst.appdata.xml [hunk 7]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<component type="desktop">
+  <id>org.kde.kst.desktop</id>
```
Comment (hfiguiere) [inline org.kde.kst.appdata.xml:3]:
```suggestion
  <id>org.kde.kst</id>
```

File: org.kde.kst.appdata.xml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  <url type="bugtracker">https://bugs.kde.org/enter_bug.cgi?product=kst&amp;format=guided</url>
+  <project_group>KDE</project_group>
+  <provides>
+    <binary>kst2</binary>
+  </provides>
+  <content_rating type="oars-1.1"/>
+  <releases>
+    <release version="2.0.81" date="2022-05-02"/>
+    <release version="2.0.8" date="2014-07-20"/>
+  </releases>
```
Comment (hfiguiere) [inline org.kde.kst.appdata.xml:23]: there is no screenshot (mandatory)

**Review by hfiguiere — CHANGES_REQUESTED** [review]
make sure the appdata file is submitted upstream (and installed)

File: flathub.json (hunk 6 above)
Comment (ripxorip) [reply flathub.json:2]: As far as I know there is no working aarch64 build of kst. I struggled a lot with adding support for the hdf5. Recent versions will not build for me on x64 either. Is support for aarch64 mandatory for publishing? If so, this pull request shall be closed and the work will be postponed for the forseable future as I have limited time to put in to this specific project.

File: org.kde.kst.appdata.xml (hunk 8 above)
Comment (ripxorip) [reply org.kde.kst.appdata.xml:23]: I am about to write a PR for gettings this merged in the upstream repo, my plan was to make a commit that removes the AppData file from here when it has been merged. Is this a blocker? I will fix the screenshots

File: org.kde.kst.appdata.xml (hunk 7 above)
Comment (ripxorip) [reply org.kde.kst.appdata.xml]: fixed

File: flathub.json (hunk 6 above)
Comment (hfiguiere) [reply flathub.json:2]: how about 
> or you apply the usual "outdated configure" fix: adding a shell source that will do `cp -p /usr/share/automake-*/config.{sub,guess} .`.

Comment (hfiguiere) [reply flathub.json:2]: there are many reason why I'd be ok for skipping aarch64, but this one ain't one.

Comment (ripxorip) [reply flathub.json:2]: This specific error or the fact that no one has compiled it for aarch64 before? 

This is my first time contributing a flatpak so "the usual outdated configure" does not tell me that much, could you give me some more pointers? Adding a shell source where?

File: org.kde.kst.yml [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+        url: 'https://github.com/tbeu/matio/archive/v1.5.21.tar.gz'
+        sha256: 2b09738fc765609794cdd4745649dcb2d7a528e9ef427c3b8e2a0d71d37d02f7
+
+  - name: hdf5
+    config-opts:
+      - '--enable-cxx'
+    sources:
+      - type: archive
+        url: https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.8/hdf5-1.8.22/src/hdf5-1.8.22.tar.gz
+        sha256: 8406d96d9355ef8961d2739fb8fd5474ad4cdf52f3cfac657733defd9709bfaa
```
Comment (hfiguiere) [inline org.kde.kst.yml:63]:
```suggestion
        sha256: 8406d96d9355ef8961d2739fb8fd5474ad4cdf52f3cfac657733defd9709bfaa
      - type: shell
        commands:
          - cp -p /usr/share/automake-*/config.{sub,guess} .
```


This is what I wrote in plain English above. And also delete the flathub.json

Comment (hfiguiere) [reply org.kde.kst.yml:63]: this assume the `config.sub` and `config.guess` are in the top level directory (they usually are)

Comment (ripxorip) [reply org.kde.kst.yml:63]: Thanks for the clarification, I have uploaded a new commit now.

**Comment by hfiguiere** [comment]
bot, build org.kde.kst

File: flathub.json (hunk 6 above)
Comment (hfiguiere) [reply flathub.json]: as for "never built", if you don't try then it will never happen.

Comment (ripxorip) [reply flathub.json]: That it true, fingers crossed for this build :)

File: org.kde.kst.yml [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+      - type: shell
+        commands:
+          - cp -p /usr/share/automake-*/config.{sub,guess} .
```
Comment (hfiguiere) [inline org.kde.kst.yml:66]:
```suggestion
          - cp -p /usr/share/automake-*/config.{sub,guess} bin
```

They put these in `bin`. *sigh*

Comment (ripxorip) [reply org.kde.kst.yml]: Thx, updated with a new commit now

Comment (ripxorip) [reply org.kde.kst.yml]: Also fixed the x64 build issue with license typo

**Comment by hfiguiere** [comment]
bot, build org.kde.kst

File: org.kde.kst.yml [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+        sha256: 95ed6ab49a0ee001255eac4e44aacb5ca4ea96ba850c08337a3e4c9a0872ccd1
+
+  - name: kst
+    buildsystem: cmake-ninja
+    config-opts:
+      - '-Dkst_release=2'
+      - '-Dkst_svnversion=0'
+      - '-Dkst_version_string=2.0.8'
+      - '-Dkst_qt5=ON'
+      - '-Dkst_install_prefix=/app'
```
Comment (Mailaender) [inline org.kde.kst.yml:93]:
```suggestion
      - '-Dkst_install_prefix=${FLATPAK_DEST}'
```

File: org.kde.kst.yml [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+      - type: git
+        url: 'https://invent.kde.org/graphics/kst-plot.git'
+        commit: 5d734f489c8f711903590ee2c9db6981c088ac3d
+      - type: patch
+        path: hdf5_include.patch
+
+  - name: kst_appstream
+    buildsystem: simple
+    build-commands:
+      - install -Dm644 org.kde.kst.appdata.xml -t /app/share/metainfo
```
Comment (Mailaender) [inline org.kde.kst.yml:106]:
```suggestion
      - install -Dm644 org.kde.kst.appdata.xml -t ${FLATPAK_DEST}/share/metainfo
```

File: org.kde.kst.yml (hunk 10 above)
Comment (ripxorip) [reply org.kde.kst.yml]: Same error with netcdf https://flathub.org/builds/#/builders/32/builds/89586, added the same shell command for that, should do the trick!

**Comment by ripxorip** [comment]
The last build failed for the same reason as for the hdf5 used to, applied the same kind of patch now

File: org.kde.kst.yml [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+        url: 'https://github.com/Unidata/netcdf-c/archive/v4.8.1.tar.gz'
+        sha256: bc018cc30d5da402622bf76462480664c6668b55eb16ba205a0dfb8647161dd0
+        x-checker-data:
+          type: anitya
+          project-id: 10354
+          stable-only: true
+          url-template: 'https://github.com/Unidata/netcdf-c/archive/v$version.tar.gz'
+      - type: shell
+        commands:
+          - cp -p /usr/share/automake-*/config.{sub,guess} bin
```
Comment (hfiguiere) [inline org.kde.kst.yml:81]: They are not in `bin` in this tarball.
```suggestion
          - cp -p /usr/share/automake-*/config.{sub,guess} .
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: org.kde.kst.yml (hunk 13 above)
Comment (ripxorip) [reply org.kde.kst.yml]: @hfiguiere Thx, updated again!

**Comment by ripxorip** [comment]
bot, build org.kde.kst

**Comment by ripxorip** [comment]
Is there a way for me to verify the test build manually? I tried queing the bot but I seem to have gotten stuck

**Comment by ripxorip** [comment]
bot, build org.kde.kst

**Comment by ripxorip** [comment]
@hfiguiere I can't believe my eyes, aarch64 bulid passed! Thanks Need to test it on an ARM device, would be awesome to run kst2 on my Pinephone :sunglasses:

**Comment by ripxorip** [comment]
bot, build org.kde.kst

File: org.kde.kst.yml [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+sdk: org.kde.Sdk
+command: kst2
+rename-desktop-file: kst2.desktop
+rename-icon: kst
+finish-args:
+  - '--share=network'
+  - '--share=ipc'
+  - '--socket=fallback-x11'
+  - '--socket=wayland'
+  - '--filesystem=home'
```
Comment (barthalion) [inline org.kde.kst.yml:13]: This doesn't seem needed – it appears to use the portal correctly on my GNOME system.

**Comment by barthalion** [comment]
```
libGL error: failed to load driver: amdgpu
libGL error: failed to open /dev/dri/card0: No such file or directory
libGL error: failed to load driver: radeonsi
```

You should add `--device=dri` to finish-args.

File: org.kde.kst.yml (hunk 14 above)
Comment (ripxorip) [reply org.kde.kst.yml:13]: Not quite sure about this one, could it break other systems if removed?

**Comment by ripxorip** [comment]
> 

Will fix

**Comment by ripxorip** [comment]
bot, build org.kde.kst

File: org.kde.kst.yml (hunk 14 above)
Comment (ripxorip) [reply org.kde.kst.yml:13]: Whats the diffrence in that regard between your GNOME system or someone elses plasma/xfce.. etc?

**Comment by nedrichards** [comment]
@flathub/kde FYI

**Comment by tsdgeos** [comment]
> @flathub/kde FYI

We usually prefer things to be in json not yaml for having the same format the other projects have (easier tooling, etc), but since kst is not part of KDE Gear I'm not going to be strict about it.

**Comment by ripxorip** [comment]
> > @flathub/kde FYI
> 
> We usually prefer things to be in json not yaml for having the same format the other projects have (easier tooling, etc), but since kst is not part of KDE Gear I'm not going to be strict about it.

Thanks for the input, I will keep that in mind the next time I find something KDE related to flatpak.

**Comment by ripxorip** [comment]
What is the next step in the process of getting this merged?

File: org.kde.kst.yml (hunk 14 above)
Comment (barthalion) [reply org.kde.kst.yml:13]: At least major distributions pull in portals implementation when installing flatpak so it should work everywhere.

**Comment by nedrichards** [comment]
Removing filesystem home to close a sandbox hole since the app is portal aware - then it's good to go.

**Comment by ripxorip** [comment]
> Removing filesystem home to close a sandbox hole since the app is portal aware - then it's good to go.

I tested the build now after removing the filesystem home switch and the now I cant access my files anymore, I am running Plasma on arch. Are the user expected to install more software to handle the portal aware features? Personally I would prefer if it was as easy as possible for the end user

**Comment by ripxorip** [comment]
@nedrichards or am I missing something? I installed the latest test build on my machine. I would have expected to be prompted to allow access to my home files or similar but nothing happened and I can't access any files.

**Comment by nedrichards** [comment]
I see that https://archlinux.org/packages/extra/x86_64/xdg-desktop-portal/ is a dependency of flatpak on arch and the kde implementation is of plasma-meta so that should be 'there' but I know that arch systems can be customised in a wide variety of ways. Having access to all of that is absolutely expected for flatpak though.

**Comment by ripxorip** [comment]
> I see that https://archlinux.org/packages/extra/x86_64/xdg-desktop-portal/ is a dependency of flatpak on arch and the kde implementation is of plasma-meta so that should be 'there' but I know that arch systems can be customised in a wide variety of ways. Having access to all of that is absolutely expected for flatpak though.

@nedrichards I tried on another similar system and it seems to work. Though I could verify that both systems have xdg-desktop-portal-kde installed along with flatpak. Looking forward to getting it merged.

**Comment by nedrichards** [comment]
/merge

btw. you won't be prompted to access your home files with portals - the way it works is that the open file dialog (which is outside your process) is clever enough to basically tunnel your consent to open a specific file into the app itself.
